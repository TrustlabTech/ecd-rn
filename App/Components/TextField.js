/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import React, { Component } from 'react'
import {
  View,
  Text,
  TextInput
} from 'react-native'

import { Colours, FontSizes } from '../GlobalStyles'

export default class TextField extends Component {

  render() {
    let label = null
    if (this.props.label) {
      label =
        (<Text style={{ padding: 5, fontSize: 16, color: Colours.primaryLowlight }}>
          { this.props.label }
        </Text>)
    }

    return (
      <View
        style={{
          paddingLeft: 5,
          paddingRight: 5,
          justifyContent: 'center',
          marginTop: 8,
          marginBottom: 8
        }}
      >
        {label}
        <View
          style={{
            borderColor: Colours.primaryLowlight,
            backgroundColor: 'white',
            borderRadius: 5,
            borderWidth: 1,
            borderStyle: 'solid',
            height: this.props.height || 50,
            width: this.props.width || null
          }}
        >

          <TextInput
            ref={(ref) => { this.textInput = ref }}
            onSubmitEditing={this.props.onSubmitEditing || null}
            returnKeyType={this.props.returnKeyType || 'done'}
            autoCapitalize={this.props.autoCapitalize || 'none'}
            keyboardType={this.props.keyboardType || 'default'}
            onChangeText={(text) => this.props.onChangeText(text) || null}
            value={this.props.value}
            placeholder={this.props.placeholder || null}
            secureTextEntry={this.props.secureTextEntry || false}
            blurOnSubmit={this.props.blurOnSubmit || false}
            maxLength={this.props.maxLength || 20}
            autoFocus={this.props.autoFocus || false}
            selectTextOnFocus={this.props.selectTextOnFocus || true}
            style={{
              marginLeft: 5,
              marginRight: 5,
              fontSize: this.props.fontSize || FontSizes.h5
            }}
          />
        </View>
      </View>
    )
  }
}

TextField.propTypes = {
  onSubmitEditing: React.PropTypes.func,
  returnKeyType: React.PropTypes.string,
  autoCapitalize: React.PropTypes.string,
  keyboardType: React.PropTypes.string,
  value: React.PropTypes.string,
  placeholder: React.PropTypes.string,
  label: React.PropTypes.string,
  secureTextEntry: React.PropTypes.bool,
  blurOnSubmit: React.PropTypes.bool,
  maxLength: React.PropTypes.number,
  autoFocus: React.PropTypes.bool,
  selectTextOnFocus: React.PropTypes.bool,
  fontSize: React.PropTypes.number,
  onChangeText: React.PropTypes.func,
  width: React.PropTypes.number,
  height: React.PropTypes.number
}
