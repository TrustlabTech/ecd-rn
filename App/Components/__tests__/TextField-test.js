/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import 'react-native'
import React from 'react'
import TextField from '../TextField'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <TextField
      value="Test Value"
      onChangeText={ (text) => ({}) }
      placeholder="Phone Number"
      maxLength={10}
      keyboardType="phone-pad"
      returnKeyType="next"
      onSubmitEditing={ () => ({}) }
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})