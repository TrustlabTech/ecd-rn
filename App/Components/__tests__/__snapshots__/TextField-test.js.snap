exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "marginBottom": 8,
      "marginTop": 8,
      "paddingLeft": 5,
      "paddingRight": 5,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#666666",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 50,
        "width": null,
      }
    }>
    <TextInput
      autoCapitalize="none"
      autoFocus={false}
      blurOnSubmit={false}
      keyboardType="phone-pad"
      maxLength={10}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Phone Number"
      returnKeyType="next"
      secureTextEntry={false}
      selectTextOnFocus={true}
      style={
        Object {
          "fontSize": 22,
          "marginLeft": 5,
          "marginRight": 5,
        }
      }
      value="Test Value" />
  </View>
</View>
`;
